include/eal.h

```diff
@@ -5,41 +5,46 @@
 #include <string>
 #include <vector>
 
+#include <gsl/gsl_interp.h>
+
 namespace eal {
 
-  enum interpolation {
-    linear,
-    spline
+  enum derivation {
+    none,
+    first,
+    second
   };
 
   nlohmann::json constructStateFromIsd(const std::string positionRotationData);
 
   std::vector<double> getPosition(std::vector<std::vector<double>> coords,
                                   std::vector<double> times,
-                                  interpolation interp, double time);
+                                  double time, const gsl_interp_type* interp);
+
   std::vector<double> getVelocity(std::vector<std::vector<double>> coords,
                                   std::vector<double> times,
-                                  interpolation interp, double time);
+                                  double time, const gsl_interp_type* interp);
```

Comment:
Since we are using GSL, I imagine it is unnecessary to redefine an enum if it already exists in gsl.

Reply:
I would prefer to keep GSL-specific enums out of our API, especially since Jesse mentioned that GSL doesn't support hermite splines, and we'll need to add that. @jessemapel What do you think? 

I was picturing the enum getting extended to something like: 

```
enum interpolation {
linear,
spline,
hermite
}
```

And maybe more!

Reply:
I agree. I consumers should not need to include the GSL headers with our library.

Reply:
This also makes it so if we end up using something other than GSL, the API does not need to change.

Reply:
Agreed.